### DISCUSSION TITLE
Is it possible to deploy Nuxt project under any subfolder without knowing the actually baseURL?

### DISCUSSION THREAD

**Discussion by DanSnow** [nuxt/nuxt#15890] [Questions]
Hi folks,

I want to deploy a Nuxt project under a subfolder. I know I can set the `cdnURL` to make assets still get an absolute path. Now my question is how can I set the `baseURL` so I can make the URL on`NuxtLink` become a relative path? Is this possible? And is this possible with SSG? Or I must use Nuxt in SSR mode?

**Edited:**

I drew a diagram to explain our architecture
![image](https://user-images.githubusercontent.com/5575082/213848555-f767e48e-e4fd-4372-9d14-d50997e0c21c.png)

**Comment by HelloAlexPan** [comment]
Following!

**Comment by ethandunford** [comment]
You can set the `baseURL` in the config 

https://nuxtjs.org/docs/configuration-glossary/configuration-router/#base

**Reply by DanSnow** [reply]
Thank you. But it seems I still need to know the full path of the subfolder. Is there a way that I don't really need to know? Maybe a method that can make all the links relative?

**Reply by ethandunford** [reply]
You might be able to write some helper functions to figure out the path on start-up.

Node has some path methods or you could use an npm library.

https://nodejs.org/api/path.html#path_path_dirname_p

https://www.npmjs.com/package/directory-paths

**Reply by DanSnow** [reply]
Do you mean by reading the filesystem? If it is, it doesn't seem to work in our case. Our site is hosted on our server, and users' servers will reverse proxy to our site with their base URL in the query string.

**Reply by ethandunford** [reply]
According to the documentation

Nuxt provides <NuxtLink> component to handle any kind of links within your application.

https://nuxt.com/docs/api/components/nuxt-link

But this doesn't seem to work locally for me 🤷

if not you might have to use regular hrefs or rethink your approach.
